exercises/phone-number/phone-number.spec.js

```diff
@@ -2,52 +2,78 @@ import PhoneNumber from './phone-number';
+    const phone = new PhoneNumber('(223) 456-7890');
+    expect(phone.number()).toEqual('2234567890');
   });
 
   xtest('cleans numbers with dots', () => {
-    const phone = new PhoneNumber('123.456.7890');
-    expect(phone.number()).toEqual('1234567890');
+    const phone = new PhoneNumber('223.456.7890');
+    expect(phone.number()).toEqual('2234567890');
   });
 
   xtest('cleans numbers with multiple spaces', () => {
-    const phone = new PhoneNumber('123 456   7890   ');
-    expect(phone.number()).toEqual('1234567890');
+    const phone = new PhoneNumber('223 456   7890   ');
+    expect(phone.number()).toEqual('2234567890');
   });
 
   xtest('invalid when 9 digits', () => {
-    const phone = new PhoneNumber('123456789');
+    const phone = new PhoneNumber('223456789');
     expect(phone.number()).toEqual(null);
   });
 
-  xtest('invalid when 11 digits', () => {
-    const phone = new PhoneNumber('21234567890');
+  xtest('invalid when 11 digits does not start with a 1', () => {
+    const phone = new PhoneNumber('22234567890');
     expect(phone.number()).toEqual(null);
   });
 
   xtest('valid when 11 digits and starting with 1', () => {
-    const phone = new PhoneNumber('11234567890');
-    expect(phone.number()).toEqual('1234567890');
+    const phone = new PhoneNumber('12234567890');
+    expect(phone.number()).toEqual('2234567890');
+  });
+
+  xtest('valid when 11 digits and starting with 1 even with punctuation', () => {
+    const phone = new PhoneNumber('+1 (223) 456-7890');
+    expect(phone.number()).toEqual('2234567890');
   });
 
   xtest('invalid when 12 digits', () => {
-    const phone = new PhoneNumber('321234567890');
+    const phone = new PhoneNumber('322234567890');
     expect(phone.number()).toEqual(null);
   });
 
   xtest('invalid with letters', () => {
-    const phone = new PhoneNumber('123-abc-7890');
+    const phone = new PhoneNumber('223-abc-7890');
     expect(phone.number()).toEqual(null);
   });
 
   xtest('invalid with punctuations', () => {
-    const phone = new PhoneNumber('123-@:!-7890');
+    const phone = new PhoneNumber('223-@:!-7890');
     expect(phone.number()).toEqual(null);
   });
 
-  xtest('invalid with right number of digits but letters mixed in', () => {
-    const phone = new PhoneNumber('1a2b3c4d5e6f7g8h9i0j');
-    expect(phone.number()).toEqual(null);
+  xtest('invalid if area code starts with 0 or 1', () => {
+    const phone1 = new PhoneNumber('(023) 456-7890');
+    const phone2 = new PhoneNumber('(123) 456-7890');
+    expect(phone1.number()).toEqual(null);
+    expect(phone2.number()).toEqual(null);
+  });
+
+  xtest('invalid if exchange code starts with 0 or 1', () => {
+    const phone1 = new PhoneNumber('(223) 056-7890');
+    const phone2 = new PhoneNumber('(223) 156-7890');
+    expect(phone1.number()).toEqual(null);
+    expect(phone2.number()).toEqual(null);
+  });
+
+  xtest('invalid when 11 digits starting with 1, ' +
```

Comment:
I see that this set of test cases is not in canonical. This check is covered under two tests above which check for correct area code and exchange code. I think these are not required.

Reply:
The following code passes all of the previous test cases. It produces invalid output such as "0234567890" and "2231567890" in these edge cases because it accepts any 11 digit number starting with 1 as valid. None of the other tests cover this possibility.

export default class PhoneNumber {

  constructor(number) {
    this.rawNumber = number;
  }

  number() {
    if (/[a-zA-Z]/.test(this.rawNumber)) {
      return null;
    }

    return this._cleanedNumber();
  }

  _cleanedNumber() {
    const num = this.rawNumber.replace(/\D/g, '');

    if (num.length === 10 && num[0] >= 2 && num[3] >= 2) {
      return num;
    }

    if (num.length === 11 && num[0] === '1') {
      return num.substr(1);
    }

    return null;
  }
}